src/Expensify.js

```diff
@@ -115,23 +115,18 @@ class Expensify extends PureComponent {
         this.appStateChangeListener = AppState.addEventListener('change', this.initializeClient);
     }
 
-    componentDidUpdate(prevProps) {
-        const previousAccountID = lodashGet(prevProps, 'session.accountID', null);
-        const currentAccountID = lodashGet(this.props, 'session.accountID', null);
-
-        if (currentAccountID && (currentAccountID !== previousAccountID)) {
-            PushNotification.register(currentAccountID);
+    componentDidUpdate() {
+        if (!(this.state.isNavigationReady || this.state.isSplashShown)) {
```

Comment:
```suggestion
        if (!this.state.isNavigationReady || !this.state.isSplashShown) {
```

Inverting a statement like this always confuses me so correct me if I have this wrong... but I _think_ we want to proceed only when **both** of these are `true`? If so, we can return early if either one is `false` and it is the same as saying both must be `true`?

fwiw, I usually have to write the possibilities out to see if I have it right - in this case I factored that if one of these values is true and one is false then the inner conditional evaluates to `true` which when inverted would be `false` and the code will run (which I think is not the correct behavior).

Reply:
Yeah taking a closer look at this you're right. My inversion was a bit off here

Reply:
I think this has been resolved?